Example of data object, transformation, and pipeline classes

dchassin opened the pull request:
The PR provide a template from which to start building the basic components.  It has a simple data object, a simple transformation, and a simple pipeline.

jongoncalves commented on source/LoadInsight.py:
```diff
@@ -0,0 +1,109 @@
+import pandas as pd
```
Need to add this lib to the `loadinsight-environment.yml`

jongoncalves commented on source/LoadInsight.py:
```diff
@@ -0,0 +1,109 @@
+	p1.run()
+
+def nop(x):
+	return(x)
```
I'm not a fan of `noop` functions like this because future cleanup can we be painful. Nonetheless, this is minor but I would leverage a `pass` here instead of a return. Better yet, we can remove the function and simply inline `lambda *args: None`.

jongoncalves commented on source/LoadInsight.py:
```diff
@@ -0,0 +1,109 @@
+				for output in outputs:
+					output.write()
+
+class Pipeline :
```
All these classes could go under the/a generics folder and the future Pipelines, Transformations and DO simply inherit from these.

jongoncalves commented on source/LoadInsight.py:
```diff
@@ -0,0 +1,109 @@
+		self.data = data
+	
+		if url != None :
+			import curl
```
we shouldn't use `curl`, `requests` is much more robust. Also, we'll move this class anyways, but I'd refrain from these style of imports and put them at the top level.

jongoncalves commented on source/LoadInsight.py:
```diff
@@ -0,0 +1,125 @@
+	remote_storage = {} # TODO: from infrastucture
+
+	def __init__(self,
+			name="NULL",
```
We should leverage python's `None` instead of hardcoding a semantically brittle `"NULL"`

dchassin commented:
I’m not worried about style now, I’m just trying to illustrate the overall structure I’d like the system to have: just a few highly flexible classes that implement the operations we need. We can throw this code away when that’s clear.

On Jun 5, 2019, at 1:42 PM, Jonathan Goncalves <notifications@github.com<mailto:notifications@github.com>> wrote:


@jongoncalves commented on this pull request.

________________________________

In source/LoadInsight.py<https://github.com/slacgismo/loadinsight/pull/2#discussion_r290925641>:

> +def nop(x):
+       return(x)
+
+def rand(x,N,M):
+       import numpy as np
+       pd.DataFrame(np.random.randn(N,M))
+
+class DataObject:
+       """DataObject hold all data artifacts as pandas DataFrames
+          and associated storage specifications.
+       """
+       local_storage = {"path":""}
+       remote_storage = {} # TODO: from infrastucture
+
+       def __init__(self,
+                       name="NULL",


We should leverage python's None instead of hardcoding a semantically brittle "NULL"

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/slacgismo/loadinsight/pull/2?email_source=notifications&email_token=AE6WRR23WGZEWFCSDKNN4PLPZAQLLA5CNFSM4HUEHRR2YY3PNVWWK3TUL52HS4DFWFIHK3DMKJSXC5LFON2FEZLWNFSXPKTDN5WW2ZLOORPWSZGOB2WW3ZY#pullrequestreview-246246887>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AE6WRRZ47ZVZSR7TI7SMNBTPZAQLLANCNFSM4HUEHRRQ>.

dchassin commented:
Restructured to follow the simpler approach.  No data object classes yet. That's just done by naming convention for now.  I'll experiment with that later.

jongoncalves commented on source/config.py:
```diff
@@ -0,0 +1,21 @@
+remote_path = "remote/"
+
+# size of an experimental data array
+size = [1000,10]
```
I dig this config file. Let's move it to a config folder and rename it dev (or test, or debug), just something to distinguish from a production type configuration. That way we can test with multiple different configuration files. Also, all these config should/could probably be all uppercase since they likely won't change inter-run

dchassin replied:
Easily done and done.

dchassin commented:
CircleCI now runs the test ok.

jongoncalves commented:
Awesome - I'll capture/reflect this approach in the AM - I'll just move the test over accordingly and make sure to leverage the pipeline you have here.

dchassin commented:
Both the selftest and LoadInsight itself are working now.  
1. I fixed the data class so it's easier to access the underlying DataFrame.
2. I added support for context information in verbose and warning messages
3. The LoadInsight RBSA pipeline runs sequentially now (although the functions are stubbed in right now)

Current issues:
1. CEUS and feeder pipelines are not done yet.
2. dependencies and concurrency in the pipeline don't work yet

dchassin commented:
I think PR#4 for `example` is the one we want. PR#2 for `simple_data_object` is probably obsolete because it did not address your comments well.

dchassin commented:
I recommend closing this PR without merging.
